snapcraft/internal/parser.py

```diff
@@ -191,12 +191,11 @@ def _process_entry(data):
     origin_dir = os.path.join(_get_base_dir(), _encode_origin(origin))
     os.makedirs(origin_dir, exist_ok=True)
 
-    class Options:
-        source = origin
-        source_type = origin_type
-
-    options = Options()
-    sources.get(origin_dir, None, options)
+    source_handler = sources.get_source_handler(origin,
+                                                source_type=origin_type)
+    handler = source_handler(origin, source_dir=origin_dir)
+    repo.check_for_command(handler.command)
```

Comment:
The problem is, the parser doesn't auto-install `build-packages` while snapcraft does and we don't need `repo.check_for_command` so tests for snapcraft require extra mocking I wasn't comfortable with just to `__init__` sources.